src/Workspaces/Core/Portable/Execution/Serialization/Checksum.cs

```diff
@@ -73,8 +75,8 @@ public override int GetHashCode()
                     hash = Hash.Combine((int)_checkSum[i], hash);
                 }
 
-                _lazyHash = hash;
-            }
+                _lazyHash = new StrongBox<int>(hash);
```

Comment:
hmmm.. I thought reference assignment is atomic in .NET. I can make it to use Interlocked.Exchange, but if what you say is right, then, there is so many bugs in compiler since compiler does what I am doing all over the places.